From 72878d247c409bb1e25874bb0e3d93b3635badb9 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:50 -0400
Subject: [misc] mei: amthif: mei_amthif_host_init: propagate errors from called functions

Message-id: <1399041470-21330-18-git-send-email-prarit@redhat.com>
Patchwork-id: 79305
O-Subject: [RHEL7.1 PATCH BZ 1088953 17/77] mei: amthif: mei_amthif_host_init: propagate errors from called functions
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit af68fb65557dee58918822a470cd64fd76ed837e
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Mon Sep 16 23:44:48 2013 +0300

    mei: amthif: mei_amthif_host_init: propagate errors from called functions

    propagate error codes from called functions, they are correct.

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/amthif.c b/drivers/misc/mei/amthif.c
index 4f259d4..d22c686 100644
--- a/drivers/misc/mei/amthif.c
+++ b/drivers/misc/mei/amthif.c
@@ -79,8 +79,10 @@ int mei_amthif_host_init(struct mei_device *dev)
 
  i = mei_me_cl_by_uuid(dev, &mei_amthif_guid);
  if (i < 0) {
-  dev_info(&dev->pdev->dev, "amthif: failed to find the client\n");
-  return -ENOENT;
+  ret = i;
+  dev_info(&dev->pdev->dev,
+   "amthif: failed to find the client %d\n", ret);
+  return ret;
  }
 
  cl->me_client_id = dev->me_clients[i].client_id;
@@ -107,8 +109,9 @@ int mei_amthif_host_init(struct mei_device *dev)
  ret = mei_cl_link(cl, MEI_IAMTHIF_HOST_CLIENT_ID);
 
  if (ret < 0) {
-  dev_err(&dev->pdev->dev, "amthif: failed link client\n");
-  return -ENOENT;
+  dev_err(&dev->pdev->dev,
+   "amthif: failed link client %d\n", ret);
+  return ret;
  }
 
  cl->state = MEI_FILE_CONNECTING;
-- 
1.7.1